extensions/amp-mega-menu/0.1/amp-mega-menu.js

```diff
@@ -182,22 +222,73 @@ export class AmpMegaMenu extends AMP.BaseElement {
       content.firstChild
     );
     content.appendChild(this.createScreenReaderCloseButton_());
-    // prevent click event listener on document from closing the menu
-    content.addEventListener('click', e => e.stopPropagation());
 
-    heading.classList.add('i-amphtml-mega-menu-heading');
     // haspopup value not set to menu since content can contain more than links.
     heading.setAttribute('aria-haspopup', 'dialog');
     heading.setAttribute('aria-controls', contentId);
-    if (!heading.hasAttribute('tabindex')) {
-      heading.setAttribute('tabindex', 0);
-    }
     heading.setAttribute('aria-expanded', 'false');
-    heading.addEventListener('click', e => this.handleHeadingClick_(e));
-    // prevent focus on mousedown so that there's no sudden shift in focus
-    // when the content container opens.
-    heading.addEventListener('mousedown', e => e.preventDefault());
-    heading.addEventListener('keydown', e => this.handleHeadingKeyDown_(e));
+  }
+
+  /**
+   * Add event listeners to all registered menu items.
+   * @private
+   */
+  registerMenuItemListeners_() {
+    this.items_.forEach(item => {
```

Comment:
Actually the DOM update listener might still be necessary since we need to add several a11y attributes when first registering each menu item. Looking at the code again, since we're adding separate event listeners to the screen reader buttons anyway, maybe it's ok to keep the existing listeners on heading elements rather than using event delegation?